bsp/boards/openmote-b/board_info.h

```diff
@@ -75,92 +75,33 @@
-    #else
-    #define PORT_delayTx                        12    //  366us (measured xxxus)
-    #endif
-    #define PORT_delayRx                         0    //    0us (can not measure)
-    // radio watchdog
-#endif
-
-//40ms slot for FSK    
-#if SLOT_TEMPLATE==42 
-    #define SLOTDURATION                        40     // ms  
-    #define PORT_TsSlotDuration                 1310   // 40ms 
+    
+typedef struct {
+    uint8_t SLOTDURATION;
+    uint8_t PORT_TsSlotDuration;
 
     // execution speed related
-    #define PORT_maxTxDataPrepare               50      // 1525 us  (based on measurement)
-    #define PORT_maxRxAckPrepare                10      // 305µs (based on measurement)
-    #define PORT_maxRxDataPrepare               10      // 305µs  (based on measurement)
-    #define PORT_maxTxAckPrepare                33      // 1000µs  (based on measurement)
-    // radio speed related
-    #define PORT_delayTx                        66      // 2000µs  (based on measurement)
-    #define PORT_delayRx                        16      // 488µs. This parameter is usually set to 0, however for Atmel on openmote-b, it takes at least 1ms for the transmission to occure because of spi delay (or other implementation specific overhead), so reciver is expected to wait a little more before turning on the radio. 
-#endif
-
-
-//40ms slot for OFDM1 MCS0-3
-#if SLOT_TEMPLATE==43 
-    #define SLOTDURATION                        40     // ms  
-    #define PORT_TsSlotDuration                 1310   // 40ms
+    // also implementation related (e.g. SoC radio or spi-connected radio because of the time to load a packet to/from radio)
+    uint8_t PORT_maxTxDataPrepare;
+    uint8_t PORT_maxRxAckPrepare;
+    uint8_t PORT_maxRxDataPrepare;
+    uint8_t PORT_maxTxAckPrepare;
 
-    // execution speed related
-    #define PORT_maxTxDataPrepare               50          //1525us (based on measurement) 
-    #define PORT_maxRxAckPrepare                10          //305us (based on measurement) 
-    #define PORT_maxRxDataPrepare               10          //305us (based on measurement) 
-    #define PORT_maxTxAckPrepare                33          //1000us (based on measurement) 
     // radio speed related
-    #define PORT_delayTx                        41          //1251us (based on measurement)  
-    #define PORT_delayRx                        16          // 488µs. This parameter is usually set to 0, however for Atmel on openmote-b, it takes at least 1ms for the transmission to occure because of spi delay (or other implementation specific overhead), so reciver is expected to wait a little more before turning on the radio. 
-#endif
-
+    // also implementation related (because the time to execute the Tx/Rx command is highly dependent on the radio AND the configuration)
+    uint8_t PORT_delayTx;                       
+    uint8_t PORT_delayRx;
+} slot_board_vars_t; //board specific slot vars
```

Comment:
Does it make sense defining this in `bsp/boards/board.h`? you could have something
like this as a default config:

```c

typedef struct {
    uint8_t SLOTDURATION;
    uint8_t PORT_TsSlotDuration;
    uint8_t PORT_maxTxDataPrepare;
    uint8_t PORT_maxRxAckPrepare;
    uint8_t PORT_maxRxDataPrepare;
    uint8_t PORT_maxTxAckPrepare;
    uint8_t PORT_delayTx;
    uint8_t PORT_delayRx;
} slot_board_vars_t;

/* This would be a default configuration */

#ifndef 10MS_SLOTDURATION
#define 10MS_SLOTDURATION             20   /* in miliseconds */
#endif
#ifndef 10MS_PORT_TsSlotDuration
#define 10MS_PORT_TsSlotDuration     655   /* 655 ticks at @32768Hz */
#endif
#ifndef 10MS_PORT_maxTxDataPrepare
#define 10MS_PORT_maxTxDataPrepare   110   /* 3355us (not measured) */
#endif
#ifndef 10MS_PORT_maxRxAckPrepare
#define 10MS_PORT_maxRxAckPrepare    20    /*  610us (not measured) */
#endif
#ifndef 10MS_PORT_maxRxDataPrepare
#define 10MS_PORT_maxRxDataPrepare   33    /* 1000us (not measured) */
#endif
#ifndef 10MS_PORT_maxTxAckPrepare
#define 10MS_PORT_maxTxAckPrepare    50    /* 1525us (not measured) */
#endif
#ifndef 10MS_PORT_delayTx
#define 10MS_PORT_delayTx            10   /*  549us (not measured) */
#endif
#ifndef 10MS_PORT_delayRx
#define 10MS_PORT_delayRx            0    /*    0us (can not measure) */
#endif

#define SLOT_10_MS_BOARD_CONFIG_DEFAULT  { .SLOTDURATION = 10MS_SLOTDURATION, \
                                           .PORT_TsSlotDuration = 10MS_PORT_TsSlotDuration, \
                                           .PORT_maxTxDataPrepare = 10MS_PORT_maxTxDataPrepare, \
                                           .PORT_maxRxAckPrepare = 10MS_PORT_maxRxAckPrepare, \
                                           .PORT_maxRxDataPrepare = 10MS_PORT_maxRxDataPrepare, \
                                           .PORT_maxTxAckPrepare = 10MS_PORT_maxTxAckPrepare, \
                                           .PORT_delayTx = 10MS_PORT_delayTx, \
                                           .PORT_delayRx = 10MS_PORT_delayRx, \
                                           }

static const slot_board_vars_t slot_10ms_board_vars[] =
{
#ifdef SLOT_10MS_BOARD_VARS
    SLOT_10MS_BOARD_VARS,
#else
    SLOT_10MS_BOARD_VARS_DEFAULT,
#endif
...
/* that for all slots */
...
#ifdef SLOT_40MS_OFDM1MCS0_3_SUBGHZ_BOARD_VARS
    SLOT_40MS_OFDM1MCS0_3_SUBGHZ_BOARD_VARS,
#else
    SLOT_40MS_OFDM1MCS0_3_SUBGHZ_BOARD_VARS_DEFAULT,
#endif
};
```

And then per `board_info` you would keep the original defines and do the following:

```c
#define SLOT_10MS_BOARD_VARS  { .SLOTDURATION = 10MS_SLOTDURATION, \
                             .PORT_TsSlotDuration =10MS_ PORT_TsSlotDuration, \
                             .PORT_maxTxDataPrepare = 10MS_PORT_maxTxDataPrepare, \
                             .PORT_maxRxAckPrepare =10MS_ PORT_maxRxAckPrepare, \
                             .PORT_maxRxDataPrepare = 10MS_PORT_maxRxDataPrepare, \
                             .PORT_maxTxAckPrepare = 10MS_PORT_maxTxAckPrepare, \
                             .PORT_delayTx = 10MS_PORT_delayTx, \
                             .PORT_delayRx = 10MS_PORT_delayRx, \
                             }
```

I think this would also make the migration easier and the init function could
simply be:

```c
void board_init_slot_vars(const slot_board_config_t *slot_board_config) {
    slot_board_vars = *slot_board_config;
}
```

The nice thing is this change could be split from this PR, and since currently only one `SLOTDURATION` is possible there would be no need to change any define per `BOARD_INFO` just add:

```
```c
#define SLOT_BOARD_VARS  { .SLOTDURATION = SLOTDURATION, \
                             .PORT_TsSlotDuration =PORT_TsSlotDuration, \
                             .PORT_maxTxDataPrepare = PORT_maxTxDataPrepare, \
                             .PORT_maxRxAckPrepare = PORT_maxRxAckPrepare, \
                             .PORT_maxRxDataPrepare = PORT_maxRxDataPrepare, \
                             .PORT_maxTxAckPrepare = PORT_maxTxAckPrepare, \
                             .PORT_delayTx = PORT_delayTx, \
                             .PORT_delayRx = PORT_delayRx, \
                             }
```
```